### ISSUE TITLE
Add another "next tab rule" for when the closed tab had just opened

### ISSUE THREAD

**Issue by pintassilgo** [mbnuqw/sidebery#474]
I'm trying to migrate from Tree Style Tab to Sidebery. After many years of development, in my opinion TST has mastered the perfect behavior about choosing *the* right tab when the current tab is closed.

Sidebery is also incredibly robust in this regard, with features like the imperative "ignore discarded tabs". But at the moment I noticed a single flaw in what I consider to be the perfect behavior.

Settings I use:
After closing current tab activate → next tab
Next tab rule → tree
Ignore folded tabs → on
Ignore discarded tabs → on
Place new tab (for the other cases) → after active tab (the same as `none` if `browser.tabs.insertAfterCurrent = true`)

There's only one case that Sidebery is handling differently (and worse in my opinion) in comparison to TST:
If you create a new tab (which opens as the next sibling tab), then close it, the previous tab should be activated (the one that was activated before creating the closed tab). Instead, Sideberry follows strictly the "new tab rule" and focus the next sibling tab.

Even that the new tab was open independently (not as a child), if I close it before switching to any other tab I would like to activate the previous tab.

Real case scenario:
I'm reading an article that mentions someone I don't know. So I open a new tab, search the name of the person and read who they are. Then I close the tab expecting to return to the article I was reading, not to switch to another tab (the next one).

**Comment by koppor** [comment]
I always wondered, why I think, Sideberry behaves strange. I think, this issue explains it.

I wonder why "previous tab" does not work. I assume, you have multiple child tabs open and begin reading from top to bottom of that "list". On close, you want to have the "next" tab opened, not the previous one.

Thus, the closing rule should be:

- In case there is a next sibling tab, focus that.
- Else: If there is a previous sibling tab, focus that.
- Else: Focus parent tab
- Else: The panel is empty

Right?

Does this solve https://github.com/mbnuqw/sidebery/issues/481?

**Comment by pintassilgo** [comment]
@koppor I disagree with your closing rules, this is not what I proposed. A new independent tab is not a child of previous tab, but still, as in my example, I want that if I close a new independent tab that just opened (i.e. before switching to another tab), previously activated tab should be activated again, not previous sibling.

What I want is mimic how Firefox works by default, without Sidebery. If you have

- Tab A (focused)
- Tab B

As per above, Tab A is focused. Then you create a new tab (like by pressing Ctrl+T), now it's like this:

- Tab A (previously focused tab)
- Tab B
- Tab C (focused)

You then close current tab, which is Tab C, by pressing Ctrl+W. Tab A will be activated, as expected.

Note this is a special rule. It's not that Firefox always activates previously active tab. The usual rule is to select previous sibling tab. The previously active tab is only selected when you close a **just-opened-tab**. A *just-opened-tab* is a new tab that never lost focus.

In my example, if after opening Tab C you switch to a different tab, then return to Tab C and close it, Tab C is no longer a *just-opened-tab*, so the "previously active tab rule" doesn't apply here, so the chosen tab after the closing will be the previous sibling (Tab B).

#481 is a different issue, it's related to Sidebery unexpectedly activating discarded tab despite user preference to skip discarded tabs.

Edit: it's worth saying all I said and all my open issues are related do Sidebery v4. They are fixed in my Sidebery v4 fork I still use today. I don't know if Sidebery v5 fixed them.
